### PR TITLE
Add tracer table to data/initialise.sql and data/initialise_local.sql

### LINKED ISSUE #1749: Make changes to enable Policy2Code AI-driven `household_net_income` analysis
@PavelMakarchuk @leehengpan

Here's what I'm thinking for a rough layout that would prevent duplicate calculation and/or simulation:

## `country.py`
* Write a utility function to take a given variable name and recurse through its `adds` and `subtracts` values and append each given variable to an array; this function will also need to handle list parameters, which can be accessed through the system’s `get_parameter` function, I believe, and can be tested for by checking if the `adds` is of type `str`
* Modify `calculate` function to
  * Set `situation.tracer` to `True`
  * Flatten `household_net_income` into an array via the above utility function
  * When calculating each variable, check if the variable is within that array and that its value isn’t equal to `variable.default_value`; if both are true, save its tracer log to the local tracers table (via `local_database`; no need to save to standard db) (more below)

## `data/initialise.sql` and `data/initialise_local.sql`
* Create a new table with roughly the following schema:
```
CREATE TABLE IF NOT EXISTS tracers (
  id INTEGER PRIMARY KEY AUTO_INCREMENT,
  household_id INT NOT NULL,
  policy_id INT NOT NULL,
  country_id VARCHAR(3) NOT NULL,
  api_version VARCHAR(10) NOT NULL,
  tracer_output SOMETHING NOT NULL (unclear what type this would be)
  variable_name VARCHAR(80) NOT NULL
)
```

## `api.py`
* Create new route to `/<country_id>/tracer` allowing for the GET method

## `endpoints/tracer.py`
* Create new controller function that
  * Pulls `household_id`, `api_version`, `variable_name`, and `policy_id` from the query params
  * Search to see if one record with all of those exist in the tracers table
  * If they do, return the `tracer_output` value and perhaps the `id` number with a 200
  * If they don’t, return a 404 and a message indicating that the record wasn’t found

### PR DESCRIPTION
**Pull request by leehengpan**
Fixes #1749

Added tracer table to `data/initialise.sql`  and `data/initialise_local.sql`.

### CODE REVIEW

**Review by anth-volk — COMMENTED** [review]
I've made a couple comments/suggestions, but so far, I think this is good progress! Thanks @leehengpan for your work on this. Feel free to re-request review whenever necessary, but might make sense to re-review when there's a controller written.

File: policyengine_api/data/initialise.sql [hunk 1]
```diff
@@ -88,4 +88,14 @@ CREATE TABLE IF NOT EXISTS user_profiles (
...
+);
+
+CREATE TABLE IF NOT EXISTS tracers (
+  id INTEGER PRIMARY KEY AUTO_INCREMENT,
+  household_id INT NOT NULL,
+  policy_id INT NOT NULL,
+  country_id VARCHAR(3) NOT NULL,
+  api_version VARCHAR(10) NOT NULL,
+  tracer_output JSON NOT NULL,
+  variable_name VARCHAR(80) NOT NULL
```
[the remark this answers is not in the document]
Comment (PavelMakarchuk) [reply policyengine_api/data/initialise.sql:100]: Actually found `mi_retirement_benefits_deduction_tier_three_ss_exempt_not_retired_eligible_people` which is 81. I can't imagine over 100 though

[the remark this answers is not in the document]
Comment (anth-volk) [reply policyengine_api/data/initialise.sql:100]: My Lord

[the remark this answers is not in the document]
Comment (leehengpan) [reply policyengine_api/data/initialise.sql]: @anth-volk Thank you for reviewing my work. I've made the change accordingly.

File: policyengine_api/data/initialise_local.sql [hunk 2]
```diff
@@ -95,4 +95,14 @@ CREATE TABLE IF NOT EXISTS user_profiles (
   username VARCHAR(255) UNIQUE,
   primary_country VARCHAR(3) NOT NULL,
   user_since BIGINT NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS tracers (
```
[the remark this answers is not in the document]
Comment (leehengpan) [reply policyengine_api/data/initialise_local.sql:101]: @anth-volk Noted. Thank you.

File: policyengine_api/country.py [hunk 3]
```diff
@@ -483,3 +488,40 @@ def validate_country(country_id: str) -> Union[None, Response]:
...
+    adds = variable.adds
+    subtracts = variable.subtracts
+    variables = []
+
+    for add in adds:
+        if isinstance(add, str):
+            variables.append(add)
+        elif isinstance(add, list):
+            for item in add:
+                variables.append(item)
```
Comment (anth-volk) [inline policyengine_api/country.py:518]: I believe your implementation here will append the entire array, but won't recurse further into `add` items further down the tree. I think you'd want to write a recursive function here that, when there is an adds and/or a subtracts, calls `get_all_variables` on that next tier downward, until eventually you hit some marker of there being no more levels.

**Review by anth-volk — CHANGES_REQUESTED** [review]
Thanks for your work on this @leehengpan. I just left a couple comments on the variable flattening utility function, as I believe this implementation may miss a couple of tiers of depth in terms of variables other variables depend upon.

**Comment by anth-volk** [comment]
Closing in light of #1782.

**Pull request closed without merging**